### PR TITLE
node 16

### PR DESCRIPTION
**Pull request by joshgubler**

### CODE REVIEW

**Comment by joshgubler** [comment]
fixes #22

**Review by stuft2 — COMMENTED** [review]
Any reason we don't just skip to Node 18?

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "lint": "npx standard index.js",
-    "package": "npx ncc build index.js -m -o dist"
+    "package": "NODE_OPTIONS=--openssl-legacy-provider npx ncc build index.js -m -o dist"
```
Comment (joshgubler) [inline package.json:8]: Updating to node18 resulted in [this error](https://github.com/byu-oit/github-action-create-ecr-repo-if-missing/actions/runs/3990547854/jobs/6844321537#step:5:13):
```
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:71:19)
    at Object.createHash (node:crypto:[13](https://github.com/byu-oit/github-action-create-ecr-repo-if-missing/actions/runs/3990547854/jobs/6844321537#step:5:14)3:10)
    at hashOf (/home/runner/work/github-action-create-ecr-repo-if-missing/github-action-create-ecr-repo-if-missing/node_modules/@zeit/ncc/dist/ncc/index.js.cache.js:3:58216)
    at module.exports (/home/runner/work/github-action-create-ecr-repo-if-missing/github-action-create-ecr-repo-if-missing/node_modules/@zeit/ncc/dist/ncc/index.js.cache.js:3:60642)
    at runCmd (/home/runner/work/github-action-create-ecr-repo-if-missing/github-action-create-ecr-repo-if-missing/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:47355)
    at 819 (/home/runner/work/github-action-create-ecr-repo-if-missing/github-action-create-ecr-repo-if-missing/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:44227)
    at __webpack_require__ (/home/runner/work/github-action-create-ecr-repo-if-missing/github-action-create-ecr-repo-if-missing/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:[16](https://github.com/byu-oit/github-action-create-ecr-repo-if-missing/actions/runs/3990547854/jobs/6844321537#step:5:17)9)
    at startup (/home/runner/work/github-action-create-ecr-repo-if-missing/github-action-create-ecr-repo-if-missing/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:339)
    at module.exports.8 (/home/runner/work/github-action-create-ecr-repo-if-missing/github-action-create-ecr-repo-if-missing/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:371)
    at /home/runner/work/github-action-create-ecr-repo-if-missing/github-action-create-ecr-repo-if-missing/node_modules/@zeit/ncc/dist/ncc/cli.js.cache.js:1:381 {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
```

I didn't get that error when running locally on the same version of Node, but a newer version of NPM. I don't see any way to specify the version of NPM in the `setup-node` action.

This `NODE_OPTIONS` was the only way I could find to get past this error. I'm open to suggestions for a better solution.

Comment (joshgubler) [reply package.json:8]: https://github.com/byu-oit/appeng-questions/discussions/17
and
https://github.com/byu-oit/appeng-questions/discussions/42
both seem related. I guess this is the best workaround for now.

**Comment by joshgubler** [comment]
This will be v1.1.2

File: action.yml [hunk 2]
```diff
@@ -20,5 +20,5 @@ inputs:
     required: false
     # Defaults to retaining all tagged images
 runs:
-  using: 'node12'
+  using: 'node18'
```
Comment (GaryGSC) [inline action.yml:23]: This [doesn't exist](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing).

The GitHub team [expects to eventually release a Node.js 20 runtime](https://github.com/actions/runner/issues/2255#issuecomment-1332128920) (`using: 'node20'`).

**Review by GaryGSC — CHANGES_REQUESTED** [review]

File: action.yml (hunk 2 above)
Comment (GaryGSC) [reply action.yml:23]: @stuft2 👆

Comment (joshgubler) [reply action.yml]: I just pushed a commit reverting to node 16.

**Comment by GaryGSC** [comment]
Thanks for fixing that. ♥

Can we call this v2? The change from `node12` to `node16` is breaking for anyone using a self-hosted runner with a version that's so old that it doesn't support `using: 'node16'`. I realize that criteria doesn't apply to BYU, but based on the handful of stars on this repo, I believe we have some external users.

**Comment by joshgubler** [comment]
Sure. Unfortunately that will be more work to get the security fixes to the end of the row. But that's life.

**Comment by GaryGSC** [comment]
Yeah. Sometimes it's hard to be good citizens. 🙁

The process would be to:
- Change the base of this PR to target `master` and update the README's example to reference `@v2`
- Merge this PR
- Cut a new `v2.0.0` release from the tip of `master`
- Make a new branch, `v2`, that matches `master` so that people can reference `@v2` and pick up minor/patch updates in the future

**Comment by GaryGSC** [comment]
(I tend to prefer branches to support the common `@v2` use case rather than floating tags.)

**Comment by GaryGSC** [comment]
@joshgubler - Do you want to handle the release process here, or would you like me to?

**Review by GaryGSC — APPROVED** [review]

**Pull request merged**